### PR TITLE
Post comments form block show warning wrt to post/page

### LINKED ISSUE #37987: Wrong notice when adding Post comment Form to a page
### Description

When you add a "Post comment Form" block in a Page, the notice showed is 
"_Post Comments Form block: comments to this post are not allowed._ " (build/block-library/index.js:30501)
and it should be 
"_Post Comments Form block: comments **to a page** are not allowed._ " 
or
"_Post Comments Form block: comments **to this post or page** are not allowed._ " 

### Step-by-step reproduction instructions

1. Go to a page
2. Opens Add a block action
3. Choose "Post comment Form"

### Screenshots, screen recording, code snippet

_No response_

### Environment info

- WordPress Multisite  Versión 5.8.3 
- Theme: Twenty Twenty Two
- Gutenberg plugin vVersión 12.3.2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by amustaque97**
Fixes: #37987

## Description

Earlier, we were showing a single hardcorded message irrespective of
post type. Warning message used to write post even in case of WP page.
It would be better if we show post in case of post and page in case of
acutal page. Merging this change will also help user in terms of reading
and understanding warning message.

## How has this been tested?

1. Checkout this branch and build the project
2. Go to a page
3. Opens Add a block action
4. Choose "Post comment Form"

Repeat above steps for create new post

## Screenshots 
**After making changes**
- Page warning
image

- Post warning
image

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/block-library/src/post-comments-form/edit.js [hunk 1]
```diff
@@ -55,8 +55,12 @@ export default function PostCommentsFormEdit( {
 
 				{ 'open' !== commentStatus && (
 					<Warning>
-						{ __(
-							'Post Comments Form block: comments to this post are not allowed.'
+						{ sprintf(
+							/* translators: 1: Post type (i.e. "post", "page") */
+							__(
+								'Post Comments Form block: comments to this %s are not allowed.'
```
Comment (alexstine) [inline packages/block-library/src/post-comments-form/edit.js:61]: @amustaque97 Isn't this also confusing? First, it mentions Post, then it mentions post type? Confusing to me anyway. Is that the actual name of the Block?

Comment (amustaque97) [reply packages/block-library/src/post-comments-form/edit.js:61]: @alexstine, yes it is the actual name of the blog. You can read here as well. https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-comments-form/block.json#L5

`Post Comments Form` is like this block is saying `some error message`... 

WDYT?

Comment (alexstine) [reply packages/block-library/src/post-comments-form/edit.js:61]: @amustaque97 One last question, should "block" be "Block"? Not sure of the capitalization requirements but would be good to keep it consistent.

Comment (paaljoachim) [reply packages/block-library/src/post-comments-form/edit.js:61]: Hey @alexstine Alex. I believe we use "block" and not "Block". I will post in the core editor channel and see what others say. I believe we are going for a more sentence kind of structure.

Comment (alexstine) [reply packages/block-library/src/post-comments-form/edit.js:61]: https://developer.wordpress.org/block-editor/explanations/user-interface/block-design/#block-identification
https://github.com/WordPress/gutenberg/issues/16118

Looks like I wasn't the only one.

**Review by alexstine — APPROVED** [review]
Looks good based on linked naming conventions.

File: packages/block-library/src/post-comments-form/edit.js (hunk 1 above)
Comment (paaljoachim) [reply packages/block-library/src/post-comments-form/edit.js:61]: Good find Alex!

**Comment by paaljoachim** [comment]
I believe that @SantosGuillamot Mario might want to take a quick look.

**Review by SantosGuillamot — APPROVED** [review]
Looks good to me as well! 🙂

**Pull request merged**